java/src/main/java/com/spotify/voyager/jni/StringIndex.java

```diff
@@ -307,9 +307,34 @@ private void assignPrimitive(List<Float> vector, float[] target) {
    *     distance from the query vector, sorted in ascending order of distance
    */
   public QueryResults query(float[] queryVector, int numNeighbors, int ef) {
-    String[] resultNames = new String[numNeighbors];
-    float[] distances = new float[numNeighbors];
-    Index.QueryResults idxResults = index.query(queryVector, numNeighbors, ef);
+    return convertResult(index.query(queryVector, numNeighbors, ef));
+  }
+
+  /**
+   * Query for multiple target vectors in parallel.
```

Comment:
I think it's worth adding a note in this docstring that this is basically just a convenience function, not a performance optimization for multiple target querying in parallel.
At some point we could potentially implement this in core as an optimization if we see a need for it

Reply:
They underlying query is in parallel though, just not the conversion from `Index.QueryResults` to `StringIndex.QueryResults`. The C++ code does each search on a separate thread. https://github.com/spotify/voyager/blob/main/cpp/TypedIndex.h#L531

Reply:
nvm I misread the code!  I see that this is actually exposing the parallelized underlying call.  Ignore me :)